packages/squiggle-lang/src/utility/SDate.ts

```diff
@@ -0,0 +1,110 @@
+import { REOther } from "../errors/messages.js";
+import { SDuration, durationUnits } from "./SDuration.js";
+import { Ok, result, Err } from "./result.js";
+import * as Result from "./result.js";
+
+function isValid(date: Date): boolean {
+  return date instanceof Date && isFinite(date.getTime());
+}
+
+function makeWithYearInt(y: number): result<SDate, string> {
+  if (y < 100) {
+    return Result.Err("Year must be over 100");
+  } else if (y > 200000) {
+    return Result.Err("Year must be less than 200000");
+  } else {
+    return Ok(new SDate(new Date(y, 0)));
+  }
+}
+
+//This is our own internal date class, which is a wrapper around the built-in Date class. It's used by the interpreter, but meant to act like a simple date library.
+export class SDate {
+  constructor(private value: Date) {
+    this.value = value;
```

Comment:
```suggestion
```

`constructor(private value: T) {}` is the shorthand for `constructor(value: T) { this.value = value; }`.

(also applies to `SDuration`)